### ISSUE TITLE
HttpWebResponse.LastModified no longer throws 

### ISSUE THREAD

**Issue by Clockwork-Muse** [dotnet/corefx#28941]
Discovered during the work for #28843

(System.Net.Request) [`HttpWebResponse.LastModified`](https://github.com/dotnet/corefx/blob/master/src/System.Net.Requests/src/System/Net/HttpWebResponse.cs#L143-L157) no longer throws `ProtocolViolationException`:

```csharp
public DateTime LastModified
{
    get
    {
        CheckDisposed();
        string lastmodHeaderValue = Headers["Last-Modified"];
        if (string.IsNullOrEmpty(lastmodHeaderValue))
        {
            return DateTime.Now;
        }
        DateTime dtOut;
        HttpDateParse.ParseHttpDate(lastmodHeaderValue, out dtOut);
        return dtOut;
    }
}
```

It looks like this occurred during the port, as [the file containing the original code was split](https://referencesource.microsoft.com/#System/net/System/Net/Internal.cs,1975), and the code throwing the exception didn't make it.  Contrast this with `HttpWebRequest.IfModifiedSince`/`Date`, [where the exception did make it through](https://github.com/dotnet/corefx/commit/9090df761a1a067c869e43066ba32fd39a26f516#diff-c70c98d319d6fa28c754b26a04324f2eR1444)

cc @davidsh

**Comment by davidsh** [comment]
Yes, this is a bug.

That code for the getter of  `HttpWebResponse.LastModified` property should be:

```c#
public DateTime LastModified
{
    get
    {
        CheckDisposed();
        string lastmodHeaderValue = Headers["Last-Modified"];
        if (string.IsNullOrEmpty(lastmodHeaderValue))
        {
            return DateTime.Now;
        }
        DateTime dtOut;
        if (HttpDateParse.ParseHttpDate(lastmodHeaderValue, out dtOut))
        {
             return dtOut;
        }
        else
        {
        throw new ProtocolViolationException(SR.net_baddate);
         }
    }
}
```

**Comment by davidsh** [comment]
cc: @karelz

**Comment by karelz** [comment]
Let's fix it in 2.1 - as compat with Framework.

**Comment by davidsh** [comment]
These are references to .NET Framework implementation

https://github.com/Microsoft/referencesource/blob/master/System/net/System/Net/HttpWebResponse.cs#L280

https://github.com/Microsoft/referencesource/blob/master/System/net/System/Net/Internal.cs#L1975
